### PR TITLE
Feature theme context

### PR DESCRIPTION
**Pull request by marcinel123**
Hi Adam,

In this PR I added custom hook which can change Light/Dark Theme.
Also corrected a few styles to see if everything changes properly.

Please have a look.
Martin

### CODE REVIEW

File: src/api/useDarkThemeMode.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { useState } from "react";
```
Comment (AdamKniec) [inline src/api/useDarkThemeMode.tsx]: Why is this file in api folder? It has nothing to do with api I guess

File: src/api/useDarkThemeMode.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+export const useDarkThemeMode = () => {
+	const [isDarkTheme, setIsDarkTheme] = useState(false);
+
+	const toggleTheme = () => {
+		setIsDarkTheme((prevState) => !prevState);
+	};
+
+	return { isDarkTheme, setIsDarkTheme, toggleTheme };
```
Comment (AdamKniec) [inline src/api/useDarkThemeMode.tsx]: maybe some memoisation here would be useful?

File: src/components/Header/Header.tsx [hunk 3]
```diff
@@ -5,12 +5,18 @@ import {
...
-			<StyledIcon className="fa-regular fa-moon"></StyledIcon>
-			Dark Mode
-		</ThemeSwitchButton>
-	</StyledHeader>
-);
+export interface HeaderProps {
+	toggleTheme: () => void;
+}
+
+export const Header = ({ toggleTheme }: HeaderProps) => {
```
Comment (AdamKniec) [inline src/components/Header/Header.tsx:12]: Why do you pass this as props? Use have a custom hook for that

Comment (AdamKniec) [reply src/components/Header/Header.tsx:12]: but It's not really important. You decide which solution to pick

File: src/api/useDarkThemeMode.tsx (hunk 1 above)
Comment (marcinel123) [reply src/api/useDarkThemeMode.tsx]: I moved it into seperate folder named "hooks" and it is inside src folder. Hope that is okay.

File: src/components/Header/Header.tsx (hunk 3 above)
Comment (marcinel123) [reply src/components/Header/Header.tsx:12]: When I tried to use toggleTheme from custom hook directly inside Header component it did not work and did not change the state.
When I pass it as props (I saw in tutorials almost always they use props in similar situation) it works correct.

File: src/api/useDarkThemeMode.tsx (hunk 2 above)
Comment (marcinel123) [reply src/api/useDarkThemeMode.tsx]: after your information. not yet with memo :)

**Pull request merged**